libtenzir/builtins/formats/yaml.cpp

```diff
@@ -425,10 +434,56 @@ class write_yaml final
   }
 };
 
+class print_yaml final : public virtual function_plugin {
+  auto name() const -> std::string override {
+    return "print_yaml";
+  }
+
+  auto make_function(invocation inv,
+                     session ctx) const -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    auto include_document_markers = std::optional<location>{};
+    auto parser = argument_parser2::function(name());
+    parser.positional("input", expr, "any");
+    parser.named("include_document_markers", include_document_markers);
+    TRY(parser.parse(inv, ctx));
+    return function_use::make(
+      [call = inv.call.get_location(), expr = std::move(expr),
+       include_document_markers](evaluator eval, session) -> multi_series {
+        return map_series(eval(expr), [&](series values) -> multi_series {
+          if (values.type.kind().is<null_type>()) {
+            return values;
+          }
+          const auto work = [&](const auto& arr) -> multi_series {
+            YAML::Emitter out;
+            auto builder = series_builder{tenzir::type{string_type{}}};
```

Comment:
This should likely use an `arrow::StringBuilder` directly.

This also applies to the other `print_*` functions.